cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -304,7 +304,8 @@ internalSetupMethod verbosity options _ bt mkargs = do
   info verbosity $ "Using internal setup method with build-type " ++ show bt
                 ++ " and args:\n  " ++ show args
   inDir (useWorkingDir options) $
-    buildTypeAction bt args
+    withEnv "HASKELL_DIST_DIR" (useDistPref options) $
```

Comment:
I'd add a comment here that in the case of the internal setup method we're doing this for the sake of build-types Configure and Make, since the Cabal library code does not care about `HASKELL_DIST_DIR`.